EIPS/eip-7069.md

```diff
@@ -0,0 +1,145 @@
+
+Furthermore, the obsolete functionality of specifying output buffer address is removed, because it is predominantly unused and implementers prefer to use `RETURNDATACOPY` instead. One exception is the case when the expected return size is known (i.e. non-dynamic return values), Solidity still uses the output buffer.
+
+Lastly, instead of returning a boolean for execution status, an extensible list of status codes is returned: `0` for success, `1` for revert, `2` for failure.
+
+We expect most new contracts to rely on the new instructions (for simplicity and in order to save gas), and some specific contracts where gas limiting is required to keep using the old instructions (e.g. [ERC-4337](./eip-4337.md)).
+
+## Motivation
+
+Observability of gas has been a problem for very long. The system of gas has been (and likely must be) flexible in adapting to changes to both how Ethereum is used as well as changes in underlying hardware.
+
+Unfortunately, in many cases compromises or workarounds had to be made to avoid affecting call instructions negatively, mostly due to the complex semantics and expectations of them.
+
+This change aims to remove gas observability from the new instructions and opening the door for new classes of contracts that are not affected by repricings. Furthermore, once the EVM Object Format (EOF) is introduced, the legacy call instructions can be rejected within EOF contracts, making sure they are mostly unaffected by changes in gas prices. Because these operations are required for removing gas observability they will be required for EOF in lieu of the existing instructions.
+
+It is important to note that starting Solidity 0.4.21, the compiler already passes all remaining gas to calls (using `call(gas(), ...`), unless the developer uses the explicit override (`{gas: ...}`) in the language. This suggests most contracts don't rely on controlling gas.
+
+Besides the above, this change introduces a convenice feature of returning more detailed status codes: success (0), revert (1), failure (2). This moves from the boolean option to codes, which are extensible in the future.
+
+Lastly, the introduction of the `RETURNDATA*` instructions ([EIP-211](./eip-211.md)) have obsoleted the output parameters of calls, mostly rendering them unused. An interesting problem to mention is the case of [ERC-20](./eip-20.md) where conflicting implementations caused a lot of trouble, where some would return something, while others wont.
+
+## Specification
+
+| Name | Value | Comment |
+|------|-------|---------|
+| WARM_STORAGE_READ_COST | 100 | From [EIP-2929](./eip-2929.md) |
+| COLD_ACCOUNT_ACCESS | 2600 | From [EIP-2929](./eip-2929.md) |
+| ACCOUNT_CREATION_COST | 25000 | |
+| MIN_RETAINED_GAS | 5000 | |
+| MIN_CALLEE_GAS | 2300 | |
+
+We introduce three new instructions:
+
+- `CALL2` (`0xf8`) with arguments `(target_address, input_offset, input_size, value)` 
```

Comment:
should preserve the original arg order from CALL:
`(target_address, value, input_offset, input_size)`

Reply:
The motivation for changing the order is that both compiler and EVM implementations can be simplified as the difference is a stack value at the top or bottom.

Reply:
may be this should be mentioned and explained in rationale for better clarity

Reply:
Added.